[Bug]: v9.7.0 and v9.7.1 break compatibility with 6000.3.13f1

### What happened?

**What happened:** Updated from v9.6.8 to v9.7.1 but unity fails to compile. Same with v9.7.0.
**Unity:** 6000.3.13f1

Compile errors with the new versions:
```
Library\PackageCache\com.coplaydev.unity-mcp@8d4c82830b6d\Runtime\Helpers\UnityPhysicsCompat.cs(49,49): error CS1069: The type name 'Physics2D' could not be found in the namespace 'UnityEngine'. This type has been forwarded to assembly 'UnityEngine.Physics2DModule, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null' Enable the built in package 'Physics 2D' in the Package Manager window to fix this error.

Library\PackageCache\com.coplaydev.unity-mcp@8d4c82830b6d\Runtime\Helpers\ScreenshotUtility.cs(269,23): error CS0103: The name 'ScreenCapture' does not exist in the current context
```

Downgrading back to v9.6.8 works

### Reproduction steps

1. Use unity 6000.3.13f1 with v9.6.8 installed and working
2. Update from v9.6.8 to v9.7.1 through package manager
3. Compile errors 

### Unity version

6000.3.13f1

### MCP for Unity package version

9.7.0

### Python server version

_No response_

### MCP client

Claude Desktop

### Transport

HTTP (default)

### OS

Windows

### Relevant logs / console output

```text

```

### Checks

- [x] I searched existing issues and did not find a duplicate
- [x] I included logs / steps to reproduce

--- a/MCPForUnity/Runtime/Helpers/ScreenshotUtility.cs
+++ b/MCPForUnity/Runtime/Helpers/ScreenshotUtility.cs
@@ -53,6 +53,7 @@ namespace MCPForUnity.Runtime.Helpers
         private static bool s_loggedLegacyScreenCaptureFallback;
         private static bool? s_screenCaptureModuleAvailable;
         private static System.Reflection.MethodInfo s_captureScreenshotMethod;
+        private static System.Reflection.MethodInfo s_captureScreenshotAsTextureMethod;
 
         /// <summary>
         /// Checks if the Screen Capture module (com.unity.modules.screencapture) is enabled.
@@ -72,6 +73,8 @@ namespace MCPForUnity.Runtime.Helpers
                     {
                         s_captureScreenshotMethod = screenCaptureType.GetMethod("CaptureScreenshot",
                             new Type[] { typeof(string), typeof(int) });
+                        s_captureScreenshotAsTextureMethod = screenCaptureType.GetMethod("CaptureScreenshotAsTexture",
+                            new Type[] { typeof(int) });
                     }
                 }
                 return s_screenCaptureModuleAvailable.Value;
@@ -249,7 +252,7 @@ namespace MCPForUnity.Runtime.Helpers
             int maxResolution = 0,
             string folderOverride = null)
         {
-            if (!IsScreenCaptureModuleAvailable)
+            if (!IsScreenCaptureModuleAvailable || s_captureScreenshotAsTextureMethod == null)
             {
                 var fallbackCamera = FindAvailableCamera();
                 if (fallbackCamera != null)
@@ -266,7 +269,9 @@ namespace MCPForUnity.Runtime.Helpers
             int imgW = 0, imgH = 0;
             try
             {
-                tex = ScreenCapture.CaptureScreenshotAsTexture(result.SuperSize);
+                // Reflective call to ScreenCapture.CaptureScreenshotAsTexture so the file compiles
+                // even when the Screen Capture module (com.unity.modules.screencapture) is disabled.
+                tex = s_captureScreenshotAsTextureMethod.Invoke(null, new object[] { result.SuperSize }) as Texture2D;
                 if (tex == null)
                 {
                     // Fallback to camera-based if ScreenCapture fails

--- a/MCPForUnity/Runtime/Helpers/UnityPhysicsCompat.cs
+++ b/MCPForUnity/Runtime/Helpers/UnityPhysicsCompat.cs
@@ -17,10 +17,15 @@ namespace MCPForUnity.Runtime.Helpers
     ///
     /// We use reflection rather than direct property access so calls stay clean of
     /// CS0618 warnings AND survive eventual removal of the obsolete property without
-    /// a recompile of this package.
+    /// a recompile of this package. Type lookups go through <see cref="Type.GetType(string)"/>
+    /// so this file compiles even when the Physics 2D built-in module is disabled in
+    /// the Package Manager.
     /// </summary>
     public static class UnityPhysicsCompat
     {
+        // Assembly-qualified name — resolved at runtime so the file compiles when the
+        // Physics 2D built-in module is disabled in the Package Manager.
+        private const string Physics2DTypeName = "UnityEngine.Physics2D, UnityEngine.Physics2DModule";
         /// <summary>
         /// Cross-version description of the 3D physics simulation mode.
         /// On 2022.2+ this maps onto <c>UnityEngine.SimulationMode</c>; on older
@@ -46,9 +51,13 @@ namespace MCPForUnity.Runtime.Helpers
                 if (!_physics2DProbed)
                 {
                     _physics2DProbed = true;
-                    _physics2DAutoSync = typeof(Physics2D).GetProperty(
-                        "autoSyncTransforms",
-                        BindingFlags.Public | BindingFlags.Static);
+                    var physics2DType = Type.GetType(Physics2DTypeName);
+                    if (physics2DType != null)
+                    {
+                        _physics2DAutoSync = physics2DType.GetProperty(
+                            "autoSyncTransforms",
+                            BindingFlags.Public | BindingFlags.Static);
+                    }
                 }
                 return _physics2DAutoSync;
             }

